adapters/repos/db/index.go

```diff
@@ -645,6 +649,7 @@ func (i *Index) updateReplicationConfig(ctx context.Context, cfg *models.Replica
 
 	i.Config.ReplicationFactor = cfg.Factor
 	i.Config.DeletionStrategy = cfg.DeletionStrategy
+	// TODO use router?
```

Comment:
Would we use the presence of the router to determine that async repl is enabled?